### PR TITLE
Improve SD XL

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR makes sure `diffusers` Stable Diffusion XL can generate images of any size. We also make sure that both single file format and diffusers format can be loaded.

**Diffusers format**:

```py
from diffusers import StableDiffusionXLPipeline, StableDiffusionXLImg2ImgPipeline
import torch

use_refiner = True

pipe = StableDiffusionXLPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-0.9", torch_dtype=torch.float16, variant="fp16", use_safetensors=True)
pipe.to("cuda")

if use_refiner:
    refiner = StableDiffusionXLImg2ImgPipeline.from_pretrained("stabilityai/stable-diffusion-xl-refiner-0.9", torch_dtype=torch.float16, use_safetensors=True, variant="fp16")
    refiner.to("cuda")

prompt = "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k"
image = pipe(prompt=prompt, output_type="latent" if use_refiner else "pil").images[0]

if use_refiner:
    image = refiner(prompt=prompt, image=image[None, :]).images[0]
```

**Single File Format**:

```py
from diffusers import StableDiffusionXLPipeline, StableDiffusionXLImg2ImgPipeline
import torch

use_refiner = True

pipe = StableDiffusionXLPipeline.from_single_file("https://huggingface.co/stabilityai/stable-diffusion-xl-base-0.9/blob/main/sd_xl_base_0.9.safetensors", torch_dtype=torch.float16, use_safetensors=True)
pipe.to("cuda")

if use_refiner:
    refiner = StableDiffusionXLImg2ImgPipeline.from_single_file("https://huggingface.co/stabilityai/stable-diffusion-xl-refiner-0.9/blob/main/sd_xl_refiner_0.9.safetensors", torch_dtype=torch.float16, use_safetensors=True)
    refiner.to("cuda")

prompt = "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k"
image = pipe(prompt=prompt, output_type="latent" if use_refiner else "pil").images[0]

if use_refiner:
    image = refiner(prompt=prompt, image=image[None, :]).images[0]
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py [hunk 1]
```diff
@@ -631,9 +630,9 @@ def __call__(
         callback_steps: int = 1,
         cross_attention_kwargs: Optional[Dict[str, Any]] = None,
         guidance_rescale: float = 0.0,
-        original_size: Tuple[int, int] = (1024, 1024),
+        original_size: Tuple[int, int] = None,
         crops_coords_top_left: Tuple[int, int] = (0, 0),
-        target_size: Tuple[int, int] = (1024, 1024),
+        target_size: Tuple[int, int] = None,
```
Comment (bghira) [inline src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py:635]: why is this now defaulting to None? can you explain a bit?

Comment (patrickvonplaten) [reply src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py:635]: It will then later be set to the passed `height` and `width` this makes sure we generate correct outputs for different input sizes than 1024

File: src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py [hunk 2]
```diff
@@ -1238,20 +1252,36 @@ def download_from_original_stable_diffusion_ckpt(
...
+            interpolation_type="linear",
+            num_train_timesteps=num_train_timesteps,
+            prediction_type="epsilon",
+            sample_max_value=1.0,
+            set_alpha_to_one=False,
+            skip_prk_steps=True,
+            steps_offset=1,
+            timestep_spacing="leading",
+        )
+        scheduler = EulerDiscreteScheduler.from_config(scheduler_dict)
```
[the remark this answers is not in the document]
Comment (JemiloII) [reply src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py:1280]: I like the idea of being able to customize the scheduler. I like Euler, but I usually customize the settings of the scheduler.

**Comment by Mark-divinci** [comment]
i want to increase the batch_size to get larger throughput,but i got linear growth of latency.if i run some incorrect settings?
